### ISSUE TITLE
Move "Framework-provided services" from "Dependency injection" to ".NET Generic Host"

### ISSUE THREAD

**Issue by Genteure** [dotnet/docs#22449]
The [Framework-provided services](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#framework-provided-services) section should be moved to the [.NET Generic Host](https://docs.microsoft.com/en-us/dotnet/core/extensions/generic-host) page since `Host` and `ConfigureServices` is not really part of Dependency Injection.
The "Framework-provided services" section would be more useful in the context of .NET Generic Host.

The examples for dependency injection should not use `Microsoft.Extensions.Hosting`, IMHO.
Hosting is built on top of DI, documents for DI should assume readers have no knowledge of Hosting.

Example for how to combine DI, Logging and Configuration by using HostBuilder would be better suited for the .NET Generic Host page.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8fd55089-b683-2202-6641-c7b61a132f14
* Version Independent ID: cc5f00d0-fb50-1229-8fa2-e66b1f45b284
* Content: [Dependency injection in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection)
* Content Source: [docs/core/extensions/dependency-injection.md](https://github.com/dotnet/docs/blob/master/docs/core/extensions/dependency-injection.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

**Comment by IEvangelist** [comment]
Hi again @Genteure 
Thanks for posting this issue.

After reading through it a few times, I'm not sure that I agree entirely. I do think that there is potential for the Generic Host article to be updated with some of the bits from the DI article, but I believe that the DI article is perfectly fine to rely on the `HostBuilder`. Console applications in .NET are best suited when using a hosting solution, without it there isn't an instance of an `IServiceCollection` and corresponding `IServiceProvider`. This comes for free with the `HostBuilder`. We have decided that is the best way to provide DI details in our docs. Since we're building out content for .NET, we prefer to rely on the existing infrastructure in place rather than building out too much conceptual articles. One could argue that discussing DI philosophically would be advantageous, but I think we really just need a bit more cross-linking between the two articles. Does that make sense?

**Comment by Genteure** [comment]
TLDR: Avoid "circular dependency documents", provides a starting point without too much noise.

---

Some backstory:
My experience with learning ASP.NET Core was ..."confusing"?
Microsoft does a great job with documentations, but sometimes documents for A mentions B and documents for B mentions A.
It was really hard to understand the full picture without being familiar with all of them in the first place.
I usually end up with 100~ browser tabs open, with 5 tabs of A, 3 tabs of B, 7 tabs of C...
I can find what I need pretty fast after working with ASP.NET Core for sometime, and have no problem understanding, but it was really really hard to begin with.

My personal opinion is to avoid "circular dependency documents" as much as possible.
Documents for something at the bottom of the stack shouldn't contain things from above.

> This comes for free with the HostBuilder.

It also come with logging and configurations. That is unnecessary for someone to use or understand DI.

This also reminded me when I was exploring Identity, I went as far as looking up source codes trying to figure out how to disable the included UI.
It's integrated seamlessly, easy to setup, easy to use until someone try to disable part of it. A bit off topic, and my memory might be off about this.

I like that `DependencyInjection`, `Logging`, and `Options` is separated and flexible, I think documentations for those should also be "separated" and flexible.

> Console applications in .NET are best suited when using a hosting solution, without it there isn't an instance of an IServiceCollection and corresponding IServiceProvider. This comes for free with the HostBuilder.

I agree partially.

There are times that HostBuilder is not really needed or can be used, or it is not a console application at all.
As a example, the project I'm working on is a WPF desktop application, with core logic in a .NET Standard project so it could be ported to console, moving away from Autofac because I don't need the features.
It has logging and configuration systems already in place. I might change the logging to `Microsoft.Extensions.Logging` but that's for another day.

```cs
using Microsoft.Extensions.DependencyInjection;
var services = new ServiceCollection();
```

Works perfectly fine.
Developers will be able to copy and run the code, it will not be conceptual only.

> One could argue that discussing DI philosophically would be advantageous

That's not the point I'm trying to make. It _could_ use some philosophy. But for me personally, I just want to know about `Microsoft.Extensions.DependencyInjection` and only `Microsoft.Extensions.DependencyInjection`.

> I think we really just need a bit more cross-linking between the two articles.

I agree, but not too much. Dependency Injection should remains Dependency Injection.

Again that's my personal opinion ;)

**Comment by Genteure** [comment]
To further clarify:

> but sometimes documents for A mentions B and documents for B mentions A.

What I meant is documents for B **depends** on A in some way.

Some link for how to use B with A in a harmonious, integrated way would really help, **after** someone understand how B works in standalone.
